### ISSUE TITLE
AutoTune RocksDB - Is it possible to autotune the database with multiple target metrics considered?

### ISSUE THREAD

**Issue by inspurwusixuan** [microsoft/nni#3042]
**Environment**:
- NNI version: 1.9
- NNI mode (local|remote|pai): local
- Client OS: Ubuntu 20.04
- Server OS (for remote mode only): N/A
- Python version: 3.6
- PyTorch/TensorFlow version: TensorFlow 1.14
- Is conda/virtualenv/venv used?: conda
- Is running in Docker?: no

**Log message**: 
 - nnimanager.log:  N/A
 - dispatcher.log: N/A
 - nnictl stdout and stderr: N/A
 

**What issue meet, what's expected?**:
Is it possible that NNI autotune the database with multiple target metrics considered? For example, in the RocksDB user case, NNI considers OPS as the target metric and wants to maximize it, but is it possible that autotune can maximize the OPS while maintaining the other target metric at some levels?

**How to reproduce it?**: 
N/A
**Additional information**:
N/A

Thank you so much for the help!!

**Comment by gxiaotian** [comment]
@inspurwusixuan There is currently no built-in tuner supporting multi-target optimization. However, it is easy to implement multi-target optimization algorithms by yourself within NNI framework. Or, you can design a single target which combines multiple targets you want to optimize and use NNI built-in tuners to tune.

**Comment by inspurwusixuan** [comment]
Thank you so much for the reply @gxiaotian. It's really helpful! But I'm not quite sure about how to design a target that wraps the multiple target metrics together and then use NNI built-in tuners to tune.  Here are my following questions:

1. Do you mean we send a multi-target result object back to the built-in tuner? 

2. Is this work with the tuner (e.g. TPE) that needs classArg (e.g. maximize, minimize) since, in my understanding, a tuner like this needs to know in what direction you want to optimize the target and it seems like it's conflict to the situation if we want to minimize a target while maximizing the other one.

3. Do you have any doc file or code file to recommend as a starting point for this? I really appreciate that!

Thanks!

**Comment by gxiaotian** [comment]
@inspurwusixuan How to warp the multiple targets into one target highly depends on your concrete demands. For example, if you want to maximize `A(x)` meanwhile minimize `B(x)` with respect to `x`, the target `C(x) = A(x) - \alpha * B(x)` may be an acceptable choice, where `\alpha` trades off the two targets. If you prefer to use built-in tuners, just manually compute `C(x)` and report it to NNI in the trial code. If you prefer to write a multi-target tuner by yourself, just report a dictionary containing `A(x)` and `B(x)` to NNI and handle the reported dictionary in the tuner you wrote.
